src/Maker/MakeUser.php

```diff
@@ -160,11 +164,17 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
         // need to write changes early so we can modify the contents below
         $generator->writeChanges();
 
+        $useAttributesForDoctrineMapping = $userClassConfiguration->isEntity() && ($this->doctrineHelper->isDoctrineSupportingAttributes()) && Kernel::VERSION_ID >= 50200 && $this->doctrineHelper->doesClassUsesAttributes($userClassNameDetails->getFullName());
+
         // B) Implement UserInterface
         $manipulator = new ClassSourceManipulator(
             $this->fileManager->getFileContents($classPath),
-            true
+            true,
+            !$useAttributesForDoctrineMapping,
+            true,
+            $useAttributesForDoctrineMapping
```

Comment:
This constructor is complex, in large part because that nasty, confusing 3rd argument (which is my fault).

Anyways, I've seen this twice so far and it gave me an idea: add a `DoctrineHelper::createClassSourceManipulator(string $className)` method that handles all of this ugliness for us.